### ISSUE TITLE
[ruff] Hex code formatting restrictions around debug f-strings are too strict on preview

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ruff#14926]
With preview disabled [playground link](https://play.ruff.rs/68e68e62-750b-411f-b997-d1948e35a661): 
```py
# input:
f"{"\xFF\N{space}"=}"
# output:
f"{"\xff\N{SPACE}"=}"

```
With preview enabled [playground link](https://play.ruff.rs/b6cb8fdb-bdd4-414e-bb49-f2de6c05b90a): 
```py
# input:
f"{"\xFF\N{space}"=}"
# output:
f"{"\xFF\N{space}"=}"

```
This change isn't observable, since the replacement of escapes happens before the debug applies:
```py
>>> f"{"\xFF\N{space}"=}"
'"ÿ "=\'ÿ \''
>>> f"{"\xff\N{SPACE}"=}"
'"ÿ "=\'ÿ \''
```
The issue with #14766 is that the change in formatting happens inside an r-string, not specifically the f-string debug. It should be fine to apply this formatting with the usual rules on string types, regardless of where the string is.
For completeness I have also previously opened the [same issue in black](https://github.com/psf/black/issues/4523), however it looks unlikely for black to have this fixed by the 2025 release, or anytime soon, as it would require resolving outstanding questions around how to handle f-string formatting.
Also cc @MichaReiser

**Comment by MichaReiser** [comment]
Thanks for reporting this and the nice write up. This is interesting. While this is a missed opportunity, I don't think it's important for us to fix it because debug expressions are rare (especially in production code), and escapes are rare, too. It's also somewhat complicated to fix because we currently don't apply any formatting for debug expressions. We just preserve them as is and I think that's a good enough default.
